Automattic/delphin
# Use Yarn to manage dependencies

This pull request updates Delphin to use [Yarn](https://yarnpkg.com/) to manage dependencies. As mentioned [here](https://code.facebook.com/posts/1840075619545360), Yarn is a fast, reliable, and secure alternative Npm client. It has several advantages over Npm such as offering an offline mode, and better network performance. Right now it divides the installation process by almost two on the first build. This process even goes from 26 seconds to less than half a second on the next builds!
#### Npm
##### First build

```
time npm install

[...]

real    2m10.855s
user    1m13.612s
sys     0m13.112s
```
##### Next builds

```
time npm install

[...]

real    0m26.348s
user    0m25.052s
sys     0m0.660s
```
#### Yarn
##### First build

```
time yarn install --force

[...]

real    1m22.335s
user    0m27.444s
sys     0m8.820s
```
##### Next builds

```
time yarn install

yarn install v0.15.1
success Already up-to-date.
Done in 0.20s.

real    0m0.412s
user    0m0.384s
sys     0m0.020s
```
#### Testing instructions
1. Run `npm install -g yarnpkg` to install Yarn
2. Run `git checkout test/yarn` and start your server
3. Check that everything works as before
#### Additional notes

CircleCi still uses `npm` for building artifacts.
#### Reviews

@Automattic/sdev-feed
